lib/ddtrace/span.rb

```diff
@@ -34,224 +29,135 @@ class Span
     # parsing 64-bit integers for distributed tracing since an upstream system may generate one
     EXTERNAL_MAX_ID = 1 << 64
 
-    # This limit is for numeric tags because uint64 could end up rounded.
-    NUMERIC_TAG_SIZE_RANGE = (-1 << 53..1 << 53).freeze
-
-    # Some associated values should always be sent as Tags, never as Metrics, regardless
-    # if their value is numeric or not.
-    # The Datadog agent will look for these values only as Tags, not Metrics.
-    # @see https://github.com/DataDog/datadog-agent/blob/2ae2cdd315bcda53166dd8fa0dedcfc448087b9d/pkg/trace/stats/aggregation.go#L13-L17
-    ENSURE_AGENT_TAGS = {
-      Ext::DistributedTracing::ORIGIN_KEY => true,
-      Ext::Environment::TAG_VERSION => true,
-      Ext::HTTP::STATUS_CODE => true,
-      Ext::NET::TAG_HOSTNAME => true
-    }.freeze
-
-    attr_accessor :name, :service, :resource, :span_type,
-                  :span_id, :trace_id, :parent_id,
-                  :status, :sampled
-
-    attr_reader \
+    attr_accessor \
+      :allocations,
+      :end_time,
+      :id,
+      :meta,
+      :metrics,
+      :name,
+      :parent_id,
+      :resource,
+      :sampled,
+      :service,
+      :type,
       :start_time,
-      :end_time
+      :status,
+      :trace_id
+
+    attr_writer \
+      :duration
 
-    attr_writer :duration
+    # For backwards compatiblity
+    # TODO: Deprecate and remove these.
+    alias :span_id :id
+    alias :span_type :type
 
     # Create a new span manually. Call the <tt>start()</tt> method to start the time
     # measurement and then <tt>stop()</tt> once the timing operation is over.
     #
     # * +service+: the service name for this span
     # * +resource+: the resource this span refers, or +name+ if it's missing.
     #     +nil+ can be used as a placeholder, when the resource value is not yet known at +#initialize+ time.
-    # * +span_type+: the type of the span (such as +http+, +db+ and so on)
+    # * +type+: the type of the span (such as +http+, +db+ and so on)
     # * +parent_id+: the identifier of the parent span
     # * +trace_id+: the identifier of the root span for this trace
+    # TODO: Remove span_type
     def initialize(
```

Comment:
It doesn't seem that we're very far from being able to call `.freeze` in the constructor. Would that be something we eventually want to get to?